code/datums/station_traits/_station_trait.dm

```diff
@@ -48,7 +48,7 @@ GLOBAL_LIST_EMPTY(lobby_station_traits)
 	if(threat_reduction)
 		GLOB.dynamic_station_traits[src] = threat_reduction
 	if(dynamic_category)
-		GLOB.dynamic_ruleset_categories = dynamic_category
+		GLOB.dynamic_ruleset_categories |= dynamic_category
```

Comment:
This won't do, because the global bitfield will still have the `RULESET_CATEGORY_DEFAULT` bitflag enabled by default, and thus the if statement will pass even for rulesets without the added flag:
https://github.com/tgstation/tgstation/blob/54068c2de574a2d9b25759d5f6cd89b56da26ef8/code/controllers/subsystem/dynamic/dynamic.dm#L946